go/ct/evm_test.go

```diff
@@ -102,3 +105,246 @@ func TestCt_ExplicitCases(t *testing.T) {
+		state.Revision = Revision(revision)
+		state.Stack = stack
+		state.BlockContext.TimeStamp = GetForkTime(state.Revision)
+
+		lfvmResultState, err := lfvmVm.StepN(state.Clone(), 1)
+		defer lfvmResultState.Release()
+
+		if err != nil {
+			t.Fatalf("failed to run test case: %v", err)
+		}
+
+		gethResultState, err := gethVm.StepN(state.Clone(), 1)
+		defer gethResultState.Release()
+
+		if err != nil {
+			t.Fatalf("failed to run test case in  VM: %v", err)
+		}
+
+		if lfvmResultState.Status != gethResultState.Status {
+			t.Fatal("invalid result, status does not match reference status:", errorReportString(state, lfvmResultState, gethResultState))
+		}
+		if lfvmResultState.Status != st.Running {
+			return
+		}
+
+		if lfvmResultState.Gas != gethResultState.Gas {
+			t.Fatal("invalid result,  gas does not match reference gas:", errorReportString(state, lfvmResultState, gethResultState))
+		}
+
+		// Hack: lfvm does a pc transformation, but for code smaller than the required jump the pc will point to a different location
+		// - Geth will point to pc+offset, whenever it is an overflow
+		// - Lfvm will point min(pc+offset, len(code))
+		if lfvmResultState.Pc == uint16(len(opCodes)) &&
+			lfvmResultState.Pc != gethResultState.Pc {
+			lfvmResultState.Pc = gethResultState.Pc
+		}
+
+		if !lfvmResultState.Eq(gethResultState) {
+			t.Fatal("invalid result,  result state does not match reference state:", lfvmResultState.Diff(gethResultState), errorReportString(state, lfvmResultState, gethResultState))
+		}
+	})
+}
+
+func fuzzVm(testee ct.Evm, f *testing.F) {
+
+	rnd := rand.New(0)
+
+	prepareInitialTestData(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+
+		if gas < 0 {
+			t.Skip("negative gas", gas)
+		}
+
+		if Revision(revision) < R07_Istanbul || Revision(revision) > NewestSupportedRevision {
+			t.Skip("unsupported revision", revision)
+		}
+
+		if len(opCodes) == 0 {
+			t.Skip("empty opCodes")
+		}
+
+		if len(opCodes) > fuzzMaximumCodeSegment {
+			t.Skip("too many opCodes,  not interesting")
+		}
+
+		// Ignore stack sizes larger than 7 words, as they are not interesting
+		// Do not ignore stack sizes close to the overflow, as they are interesting
+		if len(stackBytes) > fuzzMaximumCodeSegment*32 && len(stackBytes) < (1024-fuzzMaximumCodeSegment) {
+			t.Skip("Uninteresting stack size", len(stackBytes))
+		}
+
+		stack := st.NewStack()
+
+		for i := 0; i < len(stackBytes); i += 32 {
+			var stackValue [32]byte
+			if i+32 <= len(stackBytes) {
+				copy(stackValue[:], stackBytes[i:i+32])
+			} else {
+				copy(stackValue[:], stackBytes[i:])
+			}
+			stack.Push(NewU256FromBytes(stackValue[:]...))
+		}
+
+		code := st.NewCode(opCodes)
+		state := st.NewState(code)
+		state.Gas = vm.Gas(gas)
+		state.Revision = Revision(revision)
+		state.Stack = stack
+		state.BlockContext.TimeStamp = GetForkTime(state.Revision)
+
+		result, _ := testee.StepN(state.Clone(), 1)
```

Comment:
I guess cloning the state here is not really necessary, right?

Reply:
Second time is not necessary as long as the VM does not modify it (to keep the input for the report). 
is this the case?

Reply:
After some experiments, it seems that if I try to release the original state, there is a double release situation.
This shows me that the original state is reused during execution and for better error reporting I will keep 3 instances of the state: original state, reference result, and testee result.
This can be optimized later on if neeeded.